lib/patch-series.ts

```diff
@@ -178,42 +180,96 @@ export class PatchSeries {
             basedOn,
             cc,
             coverLetter,
-        } = PatchSeries.parsePullRequestDescription(pullRequestDescription);
+        } = await PatchSeries.parsePullRequest(workDir,
+                                               pullRequestTitle,
+                                               pullRequestBody,
+                                               senderName, senderEmail);
 
         if (basedOn && !revParse(basedOn, workDir)) {
             throw new Error(`Cannot find base branch ${basedOn}`);
         }
 
         const publishToRemote = undefined;
 
-        const project = await ProjectOptions.get(workDir, headCommit, cc || [],
+        const project = await ProjectOptions.get(workDir, headCommit, cc,
                                                  basedOn, publishToRemote,
                                                  baseCommit);
 
-        const wrapCoverLetterAtColumn = 76;
         return new PatchSeries(notes, options, project, metadata,
                                rangeDiff,
-                               md2text(coverLetter, wrapCoverLetterAtColumn),
+                               coverLetter,
                                senderName);
     }
 
-    protected static parsePullRequestDescription(description: string): {
+    protected static async parsePullRequest(workDir: string,
+                                            prTitle: string,
+                                            prBody: string,
+                                            senderName?: string,
+                                            senderEmail?: string):
+    Promise <{
+        coverLetter: string,
+        basedOn?: string,
+        cc: string[],
+    }> {
+        // Replace \r\n with \n to simplify remaining parsing.
+        // Note that md2text() in the end will do the replacement anyway.
+        prBody = prBody.replace(/\r\n/g, "\n");
+
+        // Remove template from description (if template exists)
+        try {
+            let prTemplate =
+                await git(["show",
+                           "upstream/master:.github/PULL_REQUEST_TEMPLATE.md"],
+                          { workDir });
+            // Depending on core.autocrlf setting, template may contain \r\n
+            prTemplate = prTemplate.replace(/\r\n/g, "\n");
+            prBody = prBody.replace(prTemplate, "");
+        } catch {
+        }
+
+        if (!prBody.length) {       // reject empty description
+            throw new Error("A pull request description must be provided");
+        }
+
+        let {
+            basedOn,
+            cc,
+            coverLetter,
+        } = PatchSeries.parsePullRequestBody(prBody);
+
+        // if known, add submitter to email chain
+        if (senderEmail) {
+            cc.push(`${senderName} <${senderEmail}>`);
```

Comment:
Is this the best place for this?  Two extra parms are passed.  Might be cleaner to pass the pre-populated cc array in to the method.

Reply:
Thanks for pointing out.

I figured that it can be simplified even more if I move `cc.push` block into `getFromNotes()`.

That required me to shuffle code between commits a little bit, and now `Fix footer parsing when PR body ended with newline` commit comes earlier then before.